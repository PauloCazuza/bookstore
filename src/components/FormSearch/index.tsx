import { Box, Input, Button } from "native-base";
import { useForm, Controller } from "react-hook-form";

export interface FormDataProps {
    search: string;
}

type FormSearchProps = {
    handleClick: (data) => Promise<void>
}

export default function FormSearch({ handleClick }: FormSearchProps) {
    const { control, handleSubmit } = useForm<FormDataProps>();

    return (
        <Box alignItems="center">
            <Controller
                control={control}
                name="search"
                rules={{
                    required: "Campo obrigatório"
                }}
                render={({ field: { onChange } }) => (
                    <Input
                        w="100%"
                        py="1"
                        InputRightElement={
                            <Button size="xs" rounded="none" w="1/6" h="full" onPress={handleSubmit(handleClick)} >
                                Buscar
                            </Button>
                        }
                        onChangeText={onChange}
                        placeholder="Faça sua busca aqui" />
                )}
            />
        </Box>
    )
}