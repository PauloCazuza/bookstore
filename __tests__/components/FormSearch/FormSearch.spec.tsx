import { NativeBaseProvider } from "native-base";
import renderer from "react-test-renderer";
import FormSearch from "../../../src/components/FormSearch/index";

it('renders correctly FormSearch Void Function', () => {
    const tree = renderer.create(<NativeBaseProvider><FormSearch handleClick={async () => { }} /></NativeBaseProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});