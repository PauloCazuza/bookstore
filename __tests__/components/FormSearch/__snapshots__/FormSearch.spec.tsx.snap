// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly FormSearch Void Function 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
